File: day81-90/day83/1233. Remove Sub-Folders from the Filesystem.py
from typing import List

class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort()
        ans = [folder[0]]
        for i in range(1, len(folder)):
            if folder[i].startswith(ans[-1] + "/") :
                continue
            ans.append(folder[i])
        return ans
    

sol=Solution()
folder =["/a","/a/b","/c/d","/c/d/e","/c/f","/a","/a/b"]
print(sol.removeSubfolders(folder))

File: day81-90/day83/1233. Remove Sub-Folders from the Filesystem.ts
function removeSubfolders(folder: string[]): string[] {
  folder.sort();
  let res: string[] = [folder[0]];
  for (let i = 1; i < folder.length; i++) {
    if (folder[i].startsWith(res[res.length - 1] + "/")) {
      continue;
    }
    res.push(folder[i]);
  }
  return res;
}
